dll/win32/shlwapi/utils.cpp

```diff
@@ -140,16 +140,35 @@ SHLWAPI_IsBogusHRESULT(HRESULT hr)
     return (hr == E_FAIL || hr == E_INVALIDARG || hr == E_NOTIMPL);
 }
 
+// Used for IShellFolder_GetDisplayNameOf
+struct RETRY_DATA
+{
+    UINT uRemove;
+    UINT uAdd;
+    DWORD dwRetryFlags;
+};
+static const RETRY_DATA g_RetryData[] =
+{
+    { SHGDN_FOREDITING,    0,                SFGDNO_RETRYALWAYS         },
+    { SHGDN_FORADDRESSBAR, 0,                SFGDNO_RETRYALWAYS         },
+    { 0,                   SHGDN_FORPARSING, SFGDNO_RETRYALWAYS         },
+    { SHGDN_FORPARSING,    0,                SFGDNO_RETRYWITHFORPARSING },
+    { SHGDN_INFOLDER,      0,                SFGDNO_RETRYALWAYS         },
+};
+
 /*************************************************************************
  * IShellFolder_GetDisplayNameOf [SHLWAPI.316]
+ *
+ * @note Don't confuse with <shobjidl.h> inline function of the same name.
+ *       This function retries with other flags.
  */
 EXTERN_C HRESULT WINAPI
 IShellFolder_GetDisplayNameOf(
     _In_ IShellFolder *psf,
     _In_ LPCITEMIDLIST pidl,
     _In_ DWORD uFlags,
```

Comment:
You could change the type of this parameter to `SHGDNF`, since it's what it is for the usual IShellFolder's GetDisplayNameOf() method (and is also used that way here in the code).

Reply:
Fixed.